beginner_source/dist_overview.rst

```diff
@@ -15,207 +12,76 @@ to the technology that can best serve your use case.
+- `Tensor Parallel (TP) <https://pytorch.org/docs/stable/distributed.tensor.parallel.html>`__
+- `Pipeline Parallel (PP) <https://pytorch.org/docs/main/distributed.pipelining.html>`__
+
+Sharding primitives
+*******************
+
+``DTensor`` and ``DeviceMesh`` are primitives used to build parallelism in terms of sharded or replicated tensors on N-dimensional process groups.
+
+- `DTensor <https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/README.md>`__ represents a tensor that is sharded and/or replicated, and communicates automatically to reshard tensors as needed by operations.
+- `DeviceMesh <https://pytorch.org/docs/stable/distributed.html#devicemesh>`__ abstracts the accelerator device communicators into a multi-dimensional array, which manages the underlying ``ProcessGroup``s for collective communications in multi-dimensional parallelisms.
+
+Communications APIs
+*******************
+
+The `PyTorch distributed communication layer (C10D) <https://pytorch.org/docs/stable/distributed.html>`__ offers both collective communication APIs (e.g.,
   `all_reduce <https://pytorch.org/docs/stable/distributed.html#torch.distributed.all_reduce>`__
   and `all_gather <https://pytorch.org/docs/stable/distributed.html#torch.distributed.all_gather>`__)
   and P2P communication APIs (e.g.,
   `send <https://pytorch.org/docs/stable/distributed.html#torch.distributed.send>`__
-  and `isend <https://pytorch.org/docs/stable/distributed.html#torch.distributed.isend>`__).
-  DDP and RPC (`ProcessGroup Backend <https://pytorch.org/docs/stable/rpc.html#process-group-backend>`__)
-  are built on c10d, where the former uses collective communications
-  and the latter uses P2P communications. Usually, developers do not need to
-  directly use this raw communication API, as the DDP and RPC APIs can serve
-  many distributed training scenarios. However, there are use cases where this API
-  is still helpful. One example would be distributed parameter averaging, where
-  applications would like to compute the average values of all model parameters
-  after the backward pass instead of using DDP to communicate gradients. This can
-  decouple communications from computations and allow finer-grain control over
-  what to communicate, but on the other hand, it also gives up the performance
-  optimizations offered by DDP.
+  and `isend <https://pytorch.org/docs/stable/distributed.html#torch.distributed.isend>`__),
+  which are used under the hood in all of the parallelism implementations.
   `Writing Distributed Applications with PyTorch <../intermediate/dist_tuto.html>`__
   shows examples of using c10d communication APIs.
 
+Launcher
+********
 
-Data Parallel Training
-----------------------
+``TorchElastic`` provides the widely used `torchrun <https://pytorch.org/docs/stable/elastic/run.html>` launcher script, which spawns processes on the local and remote machines for running distributed PyTorch programs.
```

Comment:
do we still want to brand it as TorchElastic? Or we should just say TorchRun?

Reply:
Cc @kurman